## импортируем библиотеку

In [1]:
import pandas

## подключаем csv.файл

In [2]:
df = pandas.read_csv("./kc_house_data.csv", encoding="latin-1")

## смотрим данные в файле

In [19]:
df.head(10)

id             date      price  bedrooms  bathrooms  sqft_living  \
0  7129300520  20141013T000000   221900.0         3       1.00         1180   
1  6414100192  20141209T000000   538000.0         3       2.25         2570   
2  5631500400  20150225T000000   180000.0         2       1.00          770   
3  2487200875  20141209T000000   604000.0         4       3.00         1960   
4  1954400510  20150218T000000   510000.0         3       2.00         1680   
5  7237550310  20140512T000000  1225000.0         4       4.50         5420   
6  1321400060  20140627T000000   257500.0         3       2.25         1715   
7  2008000270  20150115T000000   291850.0         3       1.50         1060   
8  2414600126  20150415T000000   229500.0         3       1.00         1780   
9  3793500160  20150312T000000   323000.0         3       2.50         1890   

   sqft_lot  floors  waterfront  view  ...  grade  sqft_above  sqft_basement  \
0      5650     1.0           0     0  ...      7        1180              0   
1      7242     2.0           0     0  ...      7        2170            400   
2     10000     1.0           0     0  ...      6         770              0   
3      5000     1.0           0     0  ...      7        1050            910   
4      8080     1.0           0     0  ...      8        1680              0   
5    101930     1.0           0     0  ...     11        3890           1530   
6      6819     2.0           0     0  ...      7        1715              0   
7      9711     1.0           0     0  ...      7        1060              0   
8      7470     1.0           0     0  ...      7        1050            730   
9      6560     2.0           0     0  ...      7        1890              0   

   yr_built  yr_renovated  zipcode      lat     long  sqft_living15  \
0      1955             0    98178  47.5112 -122.257           1340   
1      1951          1991    98125  47.7210 -122.319           1690   
2      1933             0    98028  47.7379 -122.233           2720   
3      1965             0    98136  47.5208 -122.393           1360   
4      1987             0    98074  47.6168 -122.045           1800   
5      2001             0    98053  47.6561 -122.005           4760   
6      1995             0    98003  47.3097 -122.327           2238   
7      1963             0    98198  47.4095 -122.315           1650   
8      1960             0    98146  47.5123 -122.337           1780   
9      2003             0    98038  47.3684 -122.031           2390   

   sqft_lot15  
0        5650  
1        7639  
2        8062  
3        5000  
4        7503  
5      101930  
6        6819  
7        9711  
8        8113  
9        7570  

[10 rows x 21 columns]

## смотрим типы данных в таблице по столбцам

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21613 entries, 0 to 21612
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   id             21613 non-null  int64  
 1   date           21613 non-null  object 
 2   price          21613 non-null  float64
 3   bedrooms       21613 non-null  int64  
 4   bathrooms      21613 non-null  float64
 5   sqft_living    21613 non-null  int64  
 6   sqft_lot       21613 non-null  int64  
 7   floors         21613 non-null  float64
 8   waterfront     21613 non-null  int64  
 9   view           21613 non-null  int64  
 10  condition      21613 non-null  int64  
 11  grade          21613 non-null  int64  
 12  sqft_above     21613 non-null  int64  
 13  sqft_basement  21613 non-null  int64  
 14  yr_built       21613 non-null  int64  
 15  yr_renovated   21613 non-null  int64  
 16  zipcode        21613 non-null  int64  
 17  lat            21613 non-null  float64
 18  long  

## уточним количество строк 
#### (ну, так.... для доп.информации)

In [5]:
df.shape

(21613, 21)

## проверим наличие нулевых строк

In [6]:
df.isnull().sum()

id               0
date             0
price            0
bedrooms         0
bathrooms        0
sqft_living      0
sqft_lot         0
floors           0
waterfront       0
view             0
condition        0
grade            0
sqft_above       0
sqft_basement    0
yr_built         0
yr_renovated     0
zipcode          0
lat              0
long             0
sqft_living15    0
sqft_lot15       0
dtype: int64

## смотрим, что по недвижимости
#### видим, что большинство квартир стоит около 645000$ (75%), имеют в своём составе 4 комнаты, 2 ванных комнаты, чаще всего это 2 этаж в доме, год постройки 1997. Квадратуру не берём, т.к. мы видим жилую площадь в доме (а это, как правило, многоквартирные дома)

In [7]:
df.describe()

id         price      bedrooms     bathrooms   sqft_living  \
count  2.161300e+04  2.161300e+04  21613.000000  21613.000000  21613.000000   
mean   4.580302e+09  5.400881e+05      3.370842      2.114757   2079.899736   
std    2.876566e+09  3.671272e+05      0.930062      0.770163    918.440897   
min    1.000102e+06  7.500000e+04      0.000000      0.000000    290.000000   
25%    2.123049e+09  3.219500e+05      3.000000      1.750000   1427.000000   
50%    3.904930e+09  4.500000e+05      3.000000      2.250000   1910.000000   
75%    7.308900e+09  6.450000e+05      4.000000      2.500000   2550.000000   
max    9.900000e+09  7.700000e+06     33.000000      8.000000  13540.000000   

           sqft_lot        floors    waterfront          view     condition  \
count  2.161300e+04  21613.000000  21613.000000  21613.000000  21613.000000   
mean   1.510697e+04      1.494309      0.007542      0.234303      3.409430   
std    4.142051e+04      0.539989      0.086517      0.766318      0.650743   
min    5.200000e+02      1.000000      0.000000      0.000000      1.000000   
25%    5.040000e+03      1.000000      0.000000      0.000000      3.000000   
50%    7.618000e+03      1.500000      0.000000      0.000000      3.000000   
75%    1.068800e+04      2.000000      0.000000      0.000000      4.000000   
max    1.651359e+06      3.500000      1.000000      4.000000      5.000000   

              grade    sqft_above  sqft_basement      yr_built  yr_renovated  \
count  21613.000000  21613.000000   21613.000000  21613.000000  21613.000000   
mean       7.656873   1788.390691     291.509045   1971.005136     84.402258   
std        1.175459    828.090978     442.575043     29.373411    401.679240   
min        1.000000    290.000000       0.000000   1900.000000      0.000000   
25%        7.000000   1190.000000       0.000000   1951.000000      0.000000   
50%        7.000000   1560.000000       0.000000   1975.000000      0.000000   
75%        8.000000   2210.000000     560.000000   1997.000000      0.000000   
max       13.000000   9410.000000    4820.000000   2015.000000   2015.000000   

            zipcode           lat          long  sqft_living15     sqft_lot15  
count  21613.000000  21613.000000  21613.000000   21613.000000   21613.000000  
mean   98077.939805     47.560053   -122.213896    1986.552492   12768.455652  
std       53.505026      0.138564      0.140828     685.391304   27304.179631  
min    98001.000000     47.155900   -122.519000     399.000000     651.000000  
25%    98033.000000     47.471000   -122.328000    1490.000000    5100.000000  
50%    98065.000000     47.571800   -122.230000    1840.000000    7620.000000  
75%    98118.000000     47.678000   -122.125000    2360.000000   10083.000000  
max    98199.000000     47.777600   -121.315000    6210.000000  871200.000000

In [11]:
df.describe(include="object")

date
count             21613
unique              372
top     20140623T000000
freq                142

## посмотрим минимальную и максимальную цены за недвижимость

In [28]:
df.price.min()

75000.0

In [24]:
df.price.max()

7700000.0

In [30]:
df.price.describe()

count    2.161300e+04
mean     5.400881e+05
std      3.671272e+05
min      7.500000e+04
25%      3.219500e+05
50%      4.500000e+05
75%      6.450000e+05
max      7.700000e+06
Name: price, dtype: float64

## средняя жилая площадь 2550 
#### (это около 20% от общей площади)

In [33]:
df.sqft_living.describe()

count    21613.000000
mean      2079.899736
std        918.440897
min        290.000000
25%       1427.000000
50%       1910.000000
75%       2550.000000
max      13540.000000
Name: sqft_living, dtype: float64

## средняя НЕ жилая площадь составляет 10688


In [34]:
df.sqft_lot.describe()

count    2.161300e+04
mean     1.510697e+04
std      4.142051e+04
min      5.200000e+02
25%      5.040000e+03
50%      7.618000e+03
75%      1.068800e+04
max      1.651359e+06
Name: sqft_lot, dtype: float64

## определим самый дорогой дом

In [94]:
df[df.price == df.price.max()]

id             date      price  bedrooms  bathrooms  \
7252  6762700020  20141013T000000  7700000.0         6        8.0   

      sqft_living  sqft_lot  floors  waterfront  view  ...  grade  sqft_above  \
7252        12050     27600     2.5           0     3  ...     13        8570   

      sqft_basement  yr_built  yr_renovated  zipcode      lat     long  \
7252           3480      1910          1987    98102  47.6298 -122.323   

      sqft_living15  sqft_lot15  
7252           3940        8800  

[1 rows x 21 columns]

## и самый дешёвый

In [67]:
df[df.price == df.price.min()]

id             date    price  bedrooms  bathrooms  sqft_living  \
1149  3421079032  20150217T000000  75000.0         1        0.0          670   

      sqft_lot  floors  waterfront  view  ...  grade  sqft_above  \
1149     43377     1.0           0     0  ...      3         670   

      sqft_basement  yr_built  yr_renovated  zipcode      lat     long  \
1149              0      1966             0    98022  47.2638 -121.906   

      sqft_living15  sqft_lot15  
1149           1160       42882  

[1 rows x 21 columns]

## определим самый старый и самый молодой дом

In [92]:
df.yr_built.min(), df.yr_built.max()

(1900, 2015)

## определим среднюю стоимость домой с 2-мя спальнями
#### и это около 401000$

In [81]:
df.groupby('bedrooms').mean()

id         price  bathrooms  sqft_living      sqft_lot  \
bedrooms                                                                     
0         4.632838e+09  4.095038e+05   0.826923  1707.846154  24141.307692   
1         4.143251e+09  3.176429e+05   1.066583   884.613065  16285.673367   
2         4.636499e+09  4.013727e+05   1.401721  1239.752899  12096.726087   
3         4.535884e+09  4.662321e+05   1.968394  1805.837235  14414.790208   
4         4.658470e+09  6.354195e+05   2.436428  2554.649666  16720.793374   
5         4.517188e+09  7.865998e+05   2.778576  3047.545284  16769.022486   
6         4.391720e+09  8.255206e+05   3.068015  3284.341912  18586.930147   
7         4.027151e+09  9.511847e+05   3.651316  3974.210526  20840.552632   
8         4.684170e+09  1.105077e+06   3.557692  3800.000000   8971.615385   
9         5.465659e+09  8.939998e+05   4.416667  3775.000000   5480.000000   
10        5.001934e+09  8.193333e+05   3.416667  3706.666667   8859.666667   
11        1.773101e+09  5.200000e+05   3.000000  3000.000000   4960.000000   
33        2.402101e+09  6.400000e+05   1.750000  1620.000000   6000.000000   

            floors  waterfront      view  condition     grade   sqft_above  \
bedrooms                                                                     
0         1.961538    0.000000  0.307692   2.846154  6.692308  1707.846154   
1         1.150754    0.025126  0.246231   3.266332  6.100503   823.859296   
2         1.306703    0.011232  0.178623   3.377899  6.871739  1107.240942   
3         1.449410    0.006515  0.187805   3.415513  7.472516  1584.442284   
4         1.616100    0.005812  0.268091   3.406568  8.116681  2198.300639   
5         1.579638    0.011868  0.409744   3.449094  8.272330  2407.566521   
6         1.606618    0.014706  0.544118   3.470588  8.080882  2494.386029   
7         1.723684    0.000000  0.500000   3.447368  8.447368  3160.263158   
8         1.692308    0.000000  0.230769   3.538462  8.076923  2660.769231   
9         2.166667    0.000000  0.000000   3.000000  7.833333  2935.000000   
10        1.666667    0.000000  0.666667   3.666667  7.666667  2456.666667   
11        2.000000    0.000000  0.000000   3.000000  7.000000  2400.000000   
33        1.000000    0.000000  0.000000   5.000000  7.000000  1040.000000   

          sqft_basement     yr_built  yr_renovated       zipcode        lat  \
bedrooms                                                                      
0              0.000000  1983.846154      0.000000  98066.692308  47.511415   
1             60.753769  1942.663317    169.638191  98103.417085  47.567169   
2            132.511957  1955.413043     81.382609  98105.953623  47.588601   
3            221.394951  1971.812602     76.001425  98076.478217  47.551478   
4            356.349026  1976.594740     87.112467  98069.749201  47.556118   
5            639.978763  1974.204872    101.061836  98070.188007  47.574169   
6            789.955882  1963.036765    168.573529  98079.463235  47.580421   
7            813.947368  1962.842105    210.657895  98085.815789  47.609024   
8           1139.230769  1956.307692      0.000000  98081.846154  47.630008   
9            840.000000  1931.666667    335.000000  98110.666667  47.652767   
10          1250.000000  1959.666667      0.000000  98038.333333  47.606700   
11           600.000000  1918.000000   1999.000000  98106.000000  47.556000   
33           580.000000  1947.000000      0.000000  98103.000000  47.687800   

                long  sqft_living15    sqft_lot15  
bedrooms                                           
0        -122.150231    2036.923077  26097.076923  
1        -122.256854    1449.567839  14240.221106  
2        -122.278570    1503.785507   9693.109058  
3        -122.213344    1835.443506  12809.343648  
4        -122.189139    2291.124818  13727.256030  
5        -122.201242    2442.706433  13089.471580  
6        -122.240728    2280.176471  14756.878676  
7        -122.240711    

## офигеть, конечно, от 33-х спального дома))
## но видим, что бОльшая часть домой - это дома с 3-мя и 4-мя спальнями

In [91]:
df.groupby('bedrooms').count()['id']

bedrooms
0       13
1      199
2     2760
3     9824
4     6882
5     1601
6      272
7       38
8       13
9        6
10       3
11       1
33       1
Name: id, dtype: int64

## десятка самых дорогих домов))

In [70]:
df.sort_values(["price"],ascending=False).iloc[0:10]

id             date      price  bedrooms  bathrooms  \
7252  6762700020  20141013T000000  7700000.0         6       8.00   
3914  9808700762  20140611T000000  7062500.0         5       4.50   
9254  9208900037  20140919T000000  6885000.0         6       7.75   
4411  2470100110  20140804T000000  5570000.0         5       5.75   
1448  8907500070  20150413T000000  5350000.0         5       5.00   
1315  7558700030  20150413T000000  5300000.0         6       6.00   
1164  1247600105  20141020T000000  5110800.0         5       5.25   
8092  1924059029  20140617T000000  4668000.0         5       6.75   
2626  7738500731  20140815T000000  4500000.0         5       5.50   
8638  3835500195  20140618T000000  4489000.0         4       3.00   

      sqft_living  sqft_lot  floors  waterfront  view  ...  grade  sqft_above  \
7252        12050     27600     2.5           0     3  ...     13        8570   
3914        10040     37325     2.0           1     2  ...     11        7680   
9254         9890     31374     2.0           0     4  ...     13        8860   
4411         9200     35069     2.0           0     0  ...     13        6200   
1448         8000     23985     2.0           0     4  ...     12        6720   
1315         7390     24829     2.0           1     4  ...     12        5000   
1164         8010     45517     2.0           1     4  ...     12        5990   
8092         9640     13068     1.0           1     4  ...     12        4820   
2626         6640     40014     2.0           1     4  ...     12        6350   
8638         6430     27517     2.0           0     0  ...     12        6430   

      sqft_basement  yr_built  yr_renovated  zipcode      lat     long  \
7252           3480      1910          1987    98102  47.6298 -122.323   
3914           2360      1940          2001    98004  47.6500 -122.214   
9254           1030      2001             0    98039  47.6305 -122.240   
4411           3000      2001             0    98039  47.6289 -122.233   
1448           1280      2009             0    98004  47.6232 -122.220   
1315           2390      1991             0    98040  47.5631 -122.210   
1164           2020      1999             0    98033  47.6767 -122.211   
8092           4820      1983          2009    98040  47.5570 -122.210   
2626            290      2004             0    98155  47.7493 -122.280   
8638              0      2001             0    98004  47.6208 -122.219   

      sqft_living15  sqft_lot15  
7252           3940        8800  
3914           3930       25449  
9254           4540       42730  
4411           3560       24345  
1448           4600       21750  
1315           4320       24619  
1164           3430       26788  
8092           3270       10454  
2626           3030       23408  
8638           3720       14592  

[10 rows x 21 columns]

## Какая в среднем общая площадь домов, у которых стоимость больше 600 000?

In [89]:
df[df.price > 600000]

id             date      price  bedrooms  bathrooms  \
3      2487200875  20141209T000000   604000.0         4       3.00   
5      7237550310  20140512T000000  1225000.0         4       4.50   
10     1736800520  20150403T000000   662500.0         3       2.50   
15     9297300055  20150124T000000   650000.0         4       3.00   
21     2524049179  20140826T000000  2000000.0         3       2.75   
...           ...              ...        ...       ...        ...   
21597   191100405  20150421T000000  1575000.0         4       3.25   
21599  7202300110  20140915T000000   810000.0         4       3.00   
21600   249000205  20141015T000000  1537000.0         5       3.75   
21605  3448900210  20141014T000000   610685.0         4       2.50   
21606  7936000429  20150326T000000  1007500.0         4       3.50   

       sqft_living  sqft_lot  floors  waterfront  view  ...  grade  \
3             1960      5000     1.0           0     0  ...      7   
5             5420    101930     1.0           0     0  ...     11   
10            3560      9796     1.0           0     0  ...      8   
15            2950      5000     2.0           0     3  ...      9   
21            3050     44867     1.0           0     4  ...      9   
...            ...       ...     ...         ...   ...  ...    ...   
21597         3410     10125     2.0           0     0  ...     10   
21599         3990      7838     2.0           0     0  ...      9   
21600         4470      8088     2.0           0     0  ...     11   
21605         2520      6023     2.0           0     0  ...      9   
21606         3510      7200     2.0           0     0  ...      9   

       sqft_above  sqft_basement  yr_built  yr_renovated  zipcode      lat  \
3            1050            910      1965             0    98136  47.5208   
5            3890           1530      2001             0    98053  47.6561   
10           1860           1700      1965             0    98007  47.6007   
15           1980            970      1979             0    98126  47.5714   
21           2330            720      1968             0    98040  47.5316   
...           ...            ...       ...           ...      ...      ...   
21597        3410              0      2007             0    98040  47.5653   
21599        3990              0      2003             0    98053  47.6857   
21600        4470              0      2008             0    98004  47.6321   
21605        2520              0      2014             0    98056  47.5137   
21606        2600            910      2009             0    98136  47.5537   

          long  sqft_living15  sqft_lot15  
3     -122.393           1360        5000  
5     -122.005           4760      101930  
10    -122.145           2210        8925  
15    -122.375           2140        4000  
21    -122.233           4110       20336  
...        ...            ...         ...  
21597 -122.223           2290       10125  
21599 -122.046           3370        6814  
21600 -122.200           2780        8964  
21605 -122.167           2520        6023  
21606 -122.398           2050        6200  

[6256 rows x 21 columns]

## а теперь посмотрим состояние домов
#### ну, собственно, не все так безоблачно. Состояние домов ниже среднего (3,4 по 5-балльной)

In [108]:
df.groupby('condition').count()['id']

condition
1       30
2      172
3    14031
4     5679
5     1701
Name: id, dtype: int64

In [87]:
df.condition.mean()

3.4094295100171195

## Выберите дом клиенту
Клиент хочет дом без подвала, с двумя этажами, стоимостью до 150000. Какая оценка по состоянию у таких домов в среднем?

In [109]:
df[
    (df['sqft_basement'] == 0) &
    (df['price'] <= 150000) &
    (df['floors'] == 2)
]

id             date     price  bedrooms  bathrooms  \
2589   5061300030  20140508T000000  134000.0         2       1.50   
10105  5466310060  20150324T000000  139500.0         2       1.50   
11255  1843200350  20140722T000000  150000.0         2       1.50   
15610  7614100080  20150211T000000  140000.0         3       1.75   
18332  9272202260  20140924T000000  130000.0         3       1.00   
18939  4239400300  20141129T000000   90000.0         3       1.00   
18988  7399100210  20141126T000000  140000.0         3       1.50   

       sqft_living  sqft_lot  floors  waterfront  view  ...  grade  \
2589           980      5000     2.0           0     0  ...      7   
10105         1230      1561     2.0           0     0  ...      7   
11255         1360      1934     2.0           0     0  ...      7   
15610         1270      8991     2.0           0     0  ...      7   
18332         1200      7000     2.0           0     0  ...      7   
18939          980      2490     2.0           0     0  ...      6   
18988         1200      2002     2.0           0     0  ...      8   

       sqft_above  sqft_basement  yr_built  yr_renovated  zipcode      lat  \
2589          980              0      1922          2003    98014  47.7076   
10105        1230              0      1983             0    98042  47.3768   
11255        1360              0      1978             0    98092  47.2857   
15610        1270              0      1981             0    98042  47.3563   
18332        1200              0      1908             0    98116  47.5883   
18939         980              0      1969             0    98092  47.3170   
18988        1200              0      1966             0    98055  47.4659   

          long  sqft_living15  sqft_lot15  
2589  -121.359           1040        5000  
10105 -122.149           1660        2243  
11255 -122.189           1360        1898  
15610 -122.149           1270        8993  
18332 -122.384           3290        6000  
18939 -122.182            980        3154  
18988 -122.189           1270        1848  

[7 rows x 21 columns]

## Выберите дом клиенту
Клиент хочет дом либо с очень красивым видом из окна, либо с видом на набережную, в очень хорошем состоянии и год постройки не меньше 1980 года. В какой ценовом диапазоне будут дома?

около 2.2 млн.$

In [122]:
df[
    (df['waterfront'] == 1) |
    (df['view'] == 1) &
    (df['condition'] == 5) &
    (df['yr_built'] >= 1980)
].describe()

id         price    bedrooms   bathrooms   sqft_living  \
count  1.630000e+02  1.630000e+02  163.000000  163.000000    163.000000   
mean   4.490512e+09  1.661876e+06    3.300613    2.677914   3173.687117   
std    2.873261e+09  1.120372e+06    1.078035    1.092456   1606.025093   
min    1.210290e+08  2.850000e+05    1.000000    0.750000    440.000000   
25%    2.013802e+09  7.600000e+05    3.000000    2.000000   2060.000000   
50%    3.761100e+09  1.400000e+06    3.000000    2.500000   2850.000000   
75%    7.377700e+09  2.215000e+06    4.000000    3.250000   3905.000000   
max    9.808701e+09  7.062500e+06    6.000000    6.750000  10040.000000   

            sqft_lot      floors  waterfront        view   condition  \
count     163.000000  163.000000       163.0  163.000000  163.000000   
mean    25371.828221    1.641104         1.0    3.766871    3.533742   
std     42944.334119    0.554191         0.0    0.561859    0.739388   
min      1767.000000    1.000000         1.0    1.000000    1.000000   
25%     10851.000000    1.000000         1.0    4.000000    3.000000   
50%     17342.000000    2.000000         1.0    4.000000    3.000000   
75%     26408.000000    2.000000         1.0    4.000000    4.000000   
max    505166.000000    3.000000         1.0    4.000000    5.000000   

            grade   sqft_above  sqft_basement     yr_built  yr_renovated  \
count  163.000000   163.000000     163.000000   163.000000    163.000000   
mean     8.773006  2473.042945     700.644172  1962.190184    512.392638   
std      1.761385  1240.439072     792.919972    27.877056    872.405490   
min      5.000000   440.000000       0.000000  1903.000000      0.000000   
25%      8.000000  1570.000000       0.000000  1940.500000      0.000000   
50%      9.000000  2180.000000     500.000000  1960.000000      0.000000   
75%     10.000000  3085.000000    1245.000000  1986.500000   1959.000000   
max     12.000000  7680.000000    4820.000000  2014.000000   2009.000000   

            zipcode         lat        long  sqft_living15     sqft_lot15  
count    163.000000  163.000000  163.000000     163.000000     163.000000  
mean   98096.527607   47.537364 -122.281601    2666.349693   22385.104294  
std       59.705586    0.115622    0.124015     823.243364   22589.013566  
min    98004.000000   47.327600 -122.514000     750.000000    2433.000000  
25%    98040.000000   47.451200 -122.376500    2080.000000   11715.000000  
50%    98075.000000   47.539500 -122.277000    2660.000000   15959.000000  
75%    98155.000000   47.606800 -122.210000    3140.000000   24765.000000  
max    98199.000000   47.772900 -122.059000    4913.000000  192268.000000

## Выберите дом клиенту
Клиент хочет дом с видом на набережную, как минимум с тремя ванными и с подвалом. Сколько вариантов есть у клиента?

68 вариантов

In [125]:
df[
    (df['view'] == 1) &
    (df['sqft_basement'] != 0) &
    (df['bathrooms'] >= 3)
]

id             date      price  bedrooms  bathrooms  \
282    7424700045  20150513T000000  2050000.0         5       3.00   
556    5486800070  20140620T000000  1950000.0         7       3.50   
779    1954700695  20140612T000000  2250000.0         5       4.25   
902    2296500136  20140509T000000   839900.0         4       3.50   
1161   7533800295  20140819T000000  1750000.0         4       3.25   
...           ...              ...        ...       ...        ...   
20535  1370800515  20141030T000000  2950000.0         4       4.25   
20676   629890070  20140515T000000   828950.0         4       3.50   
20729  3758900023  20140521T000000  1130000.0         4       3.25   
20921  3864000120  20150408T000000  1175000.0         4       3.25   
21310  2154970020  20140703T000000  2351960.0         4       4.25   

       sqft_living  sqft_lot  floors  waterfront  view  ...  grade  \
282           3830      8480     2.0           0     1  ...      9   
556           4640     15235     2.0           0     1  ...     11   
779           4860      9453     1.5           0     1  ...     10   
902           3810     13592     1.0           0     1  ...      9   
1161          3460      7749     2.0           0     1  ...     10   
...            ...       ...     ...         ...   ...  ...    ...   
20535         4470      5884     2.0           0     1  ...     11   
20676         3930      5680     2.0           0     1  ...      9   
20729         3810      8519     1.0           0     1  ...     10   
20921         3780     10099     1.0           0     1  ...     11   
21310         5010     19412     2.0           0     1  ...     11   

       sqft_above  sqft_basement  yr_built  yr_renovated  zipcode      lat  \
282          2630           1200      1905          1994    98122  47.6166   
556          2860           1780      1965          2003    98040  47.5666   
779          3100           1760      1905             0    98112  47.6196   
902          2560           1250      2013             0    98056  47.5134   
1161         3020            440      1950          1998    98115  47.6849   
...           ...            ...       ...           ...      ...      ...   
20535        3230           1240      2010             0    98199  47.6387   
20676        2820           1110      2013             0    98027  47.5528   
20729        2680           1130      2007             0    98033  47.6990   
20921        2240           1540      2006             0    98006  47.5508   
21310        4000           1010      2014             0    98040  47.5455   

          long  sqft_living15  sqft_lot15  
282   -122.287           3050        7556  
556   -122.231           3230       20697  
779   -122.286           3150        8557  
902   -122.200           3230        9311  
1161  -122.273           3030        8680  
...        ...            ...         ...  
20535 -122.405           2570        6000  
20676 -122.076           3700        5816  
20729 -122.207           3240       10748  
20921 -122.192           3120       10669  
21310 -122.211           3820       17064  

[68 rows x 21 columns]